usaspending_api/data_load/data_load_helpers.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+
+
+def capitalize_if_string(val):
+    try:
+        return val.upper()
+    except AttributeError:
+        return val
+
+
+def false_if_null(val):
+    if val is None:
+        return False
+    return val
+
+
+# TODO: replace this with cursor.morgify() in some way that doesn't need a live connection passed around everywhere
+def format_value_for_sql(val):
+    retval = val
+    if isinstance(val, str):
+        retval = "'{}'".format(val.replace("'", "''").replace('"', '""'))
+    elif val is None:
+        retval = "null"
+    elif isinstance(val, int) or isinstance(val, float):
+        retval = str(val)
+    elif isinstance(val, list):
+        if val:
+            retval = "ARRAY[{}]".format(",".join([format_value_for_sql(element) for element in val]))
+        else:
+            retval = "'{}'"
+    elif isinstance(val, datetime.datetime):
+        retval = "'{}-{}-{} {}:{}:{}'".format(val.year, val.month, val.day, val.hour, val.minute, val.second)
+
+    return retval
+
+
+def setup_mass_load_lists(load_objects, table):
+    values = []
+
+    keys = load_objects[0][table].keys()
+
+    columns = ['"{}"'.format(key) for key in load_objects[0][table].keys()]
+
+    for load_object in load_objects:
+        val = [format_value_for_sql(load_object[table][key]) for key in keys]
+        values.append(val)
+
+    col_string = "({})".format(",".join(map(str, columns)))
+    val_string = ",".join(["({})".format(",".join(map(str, value))) for value in values])
+
+    return col_string, val_string
```

Comment:
@kwhickey I don't know if that is guaranteed. A better dict method is ` .items()`, it will return the key and associated value as it iterates through the keys

Reply:
This used to be an OrderedDict or a List or something before I was assured that dicts are ordered now. Changing it back to whatever structure it was is another option

Reply:
I'm not 100% sure if an OrderedDict will guarantee Keith's question. That needs to be researched. However, I don't understand why Keith asked the question which began this comment thread. The code wasn't nor is it calling `.keys()` and `.values()` separately.

@BrianZito:
Dictionaries started to be ordered in Python 3.6 as a perk of a new implementation. Insertion order being maintained officially became a feature in Python 3.7.

We are still developing and running in Python 3.5

Reply:
Conversation ended, resolving conversation